File: consorcio/lib/domain/repositories/supply_repository.dart
import '../entities/supply.dart';

abstract class SupplyRepository {
  Future<Supply?> getByRouteCode(String routeCode);
  Future<List<Supply>> searchByPrefix(String prefix, {int limit = 20});
  Future<List<Supply>> listNear({
    required double latitude,
    required double longitude,
    required double radiusMeters,
    int limit = 250,
  });
  Future<Sed?> getSedByCode(String code);
  Future<List<Sed>> searchSedsByPrefix(String prefix, {int limit = 12});
  Future<SupplyCatalogStatus?> getCatalogStatus();
}

File: src/domain/repositories/SupplyRepository.ts
import type {
  CatalogStatusPatch,
  ParsedSupply,
  Supply,
  SupplyCatalogStatus,
} from '@/domain/entities/Supply'
import type { ParsedSed, Sed } from '@/domain/entities/Sed'

export interface SupplyRepository {
  getByRouteCode(routeCode: string): Promise<Supply | null>
  searchByPrefix(prefix: string, limit: number): Promise<Supply[]>
  listNear(
    latitude: number,
    longitude: number,
    radiusMeters: number,
    limit: number,
  ): Promise<Supply[]>
  ensureManual(input: {
    routeCode: string
    note?: string
  }): Promise<Supply>
  setLocation(
    routeCode: string,
    latitude: number,
    longitude: number,
  ): Promise<Supply>
  getSedByCode(code: string): Promise<Sed | null>
  searchSedsByPrefix(prefix: string, limit: number): Promise<Sed[]>
  upsertMany(
    supplies: ParsedSupply[],
    onProgress?: (done: number, total: number) => void,
  ): Promise<void>
  upsertSeds(
    seds: ParsedSed[],
    onProgress?: (done: number, total: number) => void,
  ): Promise<void>
  getCatalogStatus(): Promise<SupplyCatalogStatus | null>
  saveCatalogStatus(status: CatalogStatusPatch): Promise<SupplyCatalogStatus>
}
